### DISCUSSION TITLE
Filter/Scope search results in frontend

### DISCUSSION THREAD

**Discussion by naabster** [statamic/cms#5036] [Q&A] [answered]
Hi,

I want to display a frontend search form with additional filter options. I am pretty sure I can make this with Controllers and Entry Repositories, but by doing so I am not able to use search indexes.

I like to have a search index on collection "posts" for a search query string, but also be able to filter the search results.
For example have a publish_date filter or any other filter for taxonomies on the post entry.

Is something like that possible?
Thanks

**Comment by edalzell** [comment]
Not sure I understand what you're trying to do, can you give an example?

**Comment by jelleroorda** [comment, marked as answer]
No I don't think this is possible without writing your own implementation. You can still use the Search Index to get the results yourself pretty easily with something like this:

(Taken from the `Statamic\Http\Controllers\CP\SearchController`)
```php
Search::index()
            ->ensureExists()
            ->search($request->query('q'))
            ->get()
            ->filter(function ($item) {
                // Implement your own custom filtering here, for example based on publish_date
            })
            ->take(10)
            ->map(function ($item) {
                return $item->toAugmentedCollection([
                    'title', 'edit_url',
                    'collection', 'is_entry',
                    'taxonomy', 'is_term',
                    'container', 'is_asset',
                ])->withShallowNesting();
            })
 ```

**Reply by jelleroorda** [reply]
(So like you said, you'd have to make your own controller for that, or your own tag)

**Reply by naabster** [reply]
that's what I thought.

You can use (some of) the antler collection filter syntax for search index fields. So you could use transformers to populate your search index from for example taxonomies and then use the antler filters with that.
Unfortunately this is not documented very well.

**Comment by jasonvarga** [comment]
Have you tried adding condition parameters to the search tag just like the collection tag allows?

```
{{ search:results :featured:is="true" }}
```

**Reply by naabster** [reply]
Yes I tried that and it works as long as "featured" is a field on the search index.
This is fine and almost everything I need - documentation fell a bit short on this topic.

I guess it also depends on the search driver - I dont think every driver supports filtering with search. I know Algolia supports filter with search string as long as you define your search index properly.

I can totally work with a custom controller as described above!

Thanks
